server/routes/reviewer.test.js

```diff
@@ -0,0 +1,102 @@
+const request = require('supertest')
+const { appWithAllRoutes, user, reviewerUser } = require('./testutils/appSetup')
+
+const userSupplier = jest.fn()
+
+const offenderService = {
+  getOffenderNames: () => [],
+  getOffenderDetails: () => ({ displayName: 'Jimmy Choo', offenderNo: '123456' }),
+}
+
+const reviewService = {
+  getStatements: jest.fn(),
+  getReports: () => ({ awaiting: [], completed: [] }),
+  getReport: jest.fn(),
+}
+
+const reportDetailBuilder = {
+  build: jest.fn().mockResolvedValue({ id: 1, form: { incidentDetails: {} } }),
+}
+
+let app
+
+beforeEach(() => {
+  userSupplier.mockReturnValue(user)
+  reviewService.getReport.mockResolvedValue({})
+  app = appWithAllRoutes({ offenderService, reviewService, reportDetailBuilder }, userSupplier)
+})
+
+afterEach(() => {
+  jest.resetAllMocks()
+})
+
+describe('GET /all-incidents', () => {
+  it('should render page for reviewer', () => {
+    userSupplier.mockReturnValue(reviewerUser)
+
+    return request(app)
+      .get('/all-incidents')
+      .expect(200)
+      .expect('Content-Type', /html/)
+      .expect(res => {
+        expect(res.text).toContain('Use of force incidents')
+      })
+  })
+
+  it('should not allow if not reviewer', () => {
+    userSupplier.mockReturnValue(user)
+    return request(app)
+      .get('/all-incidents')
+      .expect(401)
+      .expect(res => expect(res.text).toContain('Not authorised to access this resource'))
+  })
+})
+
+describe('GET /view-report', () => {
+  it('should render page for reviewer', () => {
+    userSupplier.mockReturnValue(reviewerUser)
+    reviewService.getReport.mockReturnValue({ id: 1, form: { incidentDetails: {} } })
+    return request(app)
+      .get('/1/view-report')
+      .expect(200)
+      .expect('Content-Type', /html/)
+      .expect(res => {
+        expect(res.text).toContain('Use of force report')
+        expect(res.text).not.toContain('Delete report')
+      })
+  })
```

Comment:
Is this missing the `should fail to render page when no report` test?

Reply:
the logic has been pushed down to the service - so we don't explicitly need to test it at this level, its just an unexpected error